### DISCUSSION TITLE
How to configure indexer when we use opengrok-docker image

### DISCUSSION THREAD

**Discussion by jkjha** [oracle/opengrok#4085] [Q&A]
I am moving my opengrok infrastructure from VM to docker and following the link mostly mentioned here: https://santhoshkmunigala.medium.com/5-steps-to-setup-opengrok-on-your-computer-a41b43af05d5

Based on above link and other official documentations, it seems auto indexing etc are configured automatically in the image. So, I placed helloworld git project under /opengrok/src folder. But it never seems to index it (/opengrok/data is always empty)... also I see no project list dropdown on UI: 

<img width="1189" alt="image" src="https://user-images.githubusercontent.com/4188259/199225179-279b29d2-554a-43de-91b3-e2fe50f93d86.png">

Can anyone please help what I might be missing here.

Some other info:

I tried accessing this link but it returns 401 ... https://example.com/api/v1/projects/indexed 

Any help highly appreciated.

**Comment by vladak** [comment]
How do you start the container ? What OpenGrok version do you use ? What do the logs say ?

As for the API request, you need to either setup token bearer authentication (see the REST related tunables on https://github.com/oracle/opengrok/blob/master/docker/README.md#environment-variables) or invoke the API request from within the container, like so:
```
docker exec <INSERT_THE_NAME_OF_THE_CONTAINER_HERE> curl -s http://localhost:8080/api/v1/projects/indexed
```
There is a good reason for that - some people might want to set up OpenGrok with authorization and exposing project metadata would be considered info leak in such setup.

**Comment by jkjha** [comment]
My issue was project list not getting displayed on UI. I understand now that the /api/v1/projects/indexed link is accessible only from localhost. 
I fixed project not getting displayed issue by:

1. Extracting [opengrok-1.7.37.tar.gz](https://github.com/oracle/opengrok/releases/download/1.7.37/opengrok-1.7.37.tar.gz) file at /opengrok/dist folder.
2. Then going inside the container and running this command (since localhost:8080 is accessible only from the container):

`java -Djava.util.logging.config.file=/opengrok/etc/logging.properties -jar /opengrok/dist/lib/opengrok.jar -c /usr/local/bin/ctags -s /opengrok/src -d /opengrok/data -H -P -S -G -W /opengrok/etc/configuration.xml -U http://localhost:8080`

After running above command, I am able to see projects list etc on the UI. So quick question, is this the correct approach (manual indexing) or container takes care of auto indexing which I am missing?

**Reply by vladak** [reply]
The idea of the Docker container is that everything is packaged inside. It's a little virtual OpenGrok appliance. Running the indexer in the container directly is not recommended. The container should take care of this. Again, you need to inspect the logs etc.

**Reply by jkjha** [reply]
thanks, I will try to look into this more.
